zhaquirks/siglis/zigfred.py

```diff
@@ -0,0 +1,267 @@
+"""zigfred device handler."""
+import logging
+from typing import Any, List, Optional, Union
+
+from zigpy.profiles import zha
+from zigpy.quirks import CustomCluster, CustomDevice
+import zigpy.types as t
+from zigpy.zcl import foundation
+from zigpy.zcl.clusters.general import (
+    Basic,
+    GreenPowerProxy,
+    Groups,
+    Identify,
+    LevelControl,
+    OnOff,
+    Scenes,
+)
+from zigpy.zcl.clusters.lighting import Color
+
+from zhaquirks.const import (
+    BUTTON,
+    BUTTON_1,
+    BUTTON_2,
+    BUTTON_3,
+    BUTTON_4,
+    COMMAND,
+    DEVICE_TYPE,
+    DOUBLE_PRESS,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    LONG_PRESS,
+    LONG_RELEASE,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PRESS_TYPE,
+    PROFILE_ID,
+    SHORT_PRESS,
+    ZHA_SEND_EVENT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+# Siglis zigfred specific clusters
+SIGLIS_MANUFACTURER_CODE = 0x129C
+ZIGFRED_CLUSTER_ID = 0xFC42
+ZIGFRED_CLUSTER_BUTTONS_ATTRIBUTE_ID = 0x0008
+ZIGFRED_CLUSTER_COMMAND_BUTTON_EVENT = 0x02
+
+
+# Siglis zigfred cluster implementation
+class ZigfredCluster(CustomCluster):
+    """Siglis manufacturer specific cluster for zigfred."""
+
+    name = "Siglis Manufacturer Specific"
+    cluster_id = ZIGFRED_CLUSTER_ID
+    buttons_attribute_id = ZIGFRED_CLUSTER_BUTTONS_ATTRIBUTE_ID
+
+    manufacturer_server_commands = {
+        ZIGFRED_CLUSTER_COMMAND_BUTTON_EVENT: ("button_event", (t.uint32_t,), False),
+    }
+
+    # manufacturer_attributes = {
+    #    buttons_attribute_id: ("buttons", t.uint32_t),
+    # }
+
+    def _process_button_event(self, value: t.uint32_t):
+        button_lookup = {
+            0: BUTTON_1,
+            1: BUTTON_2,
+            2: BUTTON_3,
+            3: BUTTON_4,
+        }
+
+        press_type_lookup = {
+            0: LONG_RELEASE,
+            1: SHORT_PRESS,
+            2: DOUBLE_PRESS,
+            3: LONG_PRESS,
+        }
+
+        button = value & 0xFF
+        press_type = (value >> 8) & 0xFF
+
+        button = button_lookup[button]
+        press_type = press_type_lookup[press_type]
+
+        action = f"{button}_{press_type}"
+
+        event_args = {
+            BUTTON: button,
+            PRESS_TYPE: press_type,
+        }
+
+        _LOGGER.info(f"Got button press on zigfred cluster: {action}")
+
+        if button and press_type:
+            self.listener_event(ZHA_SEND_EVENT, action, event_args)
+
+    def handle_cluster_request(
+        self,
+        hdr: foundation.ZCLHeader,
+        args: List[Any],
+        *,
+        dst_addressing: Optional[
+            Union[t.Addressing.Group, t.Addressing.IEEE, t.Addressing.NWK]
+        ] = None,
+    ) -> None:
+        """Handle cluster specific commands."""
+        if hdr.command_id == ZIGFRED_CLUSTER_COMMAND_BUTTON_EVENT:
+            self._process_button_event(args[0])
+
+    # def _update_attribute(self, attrid, value):
```

Comment:
Can you remove the commented out code please?

Reply:
Thanks @dmulcahey
I just did a pull request with the commented code removed.